src/player.cpp

```diff
@@ -3469,8 +3468,7 @@ void Player::onWalkAborted()
 void Player::onWalkComplete()
 {
 	if (walkTask) {
-		walkTaskEvent = g_scheduler.addEvent(walkTask);
-		walkTask = nullptr;
+		walkTaskEvent = g_scheduler.addEvent(std::move(walkTask));
```

Comment:
Prone to use-after-move, you still need `walkTask = nullptr`

Reply:
I think your suggestion is good, good capture, I was thinking about this:

```cpp
walkTaskEvent = g_scheduler.addEvent(SchedulerTask_ptr(walkTask.release()));
```

However, I am not convinced about having to build a new unique_ptr, although it should be free, I don't know how profitable it is.

The option to assign nullptr is correct and I like it